src/main/java/org/checkerframework/specimin/SpeciminRunner.java

```diff
@@ -155,54 +172,111 @@ private static void performMinimizationImpl(
+       >>> If JAR paths are provided (jarPaths list is not empty), it decompiles the JAR files to Java source files in the root directory using the ConsoleDecompiler tool.
+     */
     if (!jarPaths.isEmpty()) {
       List<String> argsToDecompile = new ArrayList<>();
-      argsToDecompile.add("--silent");
-      argsToDecompile.addAll(jarPaths);
-      argsToDecompile.add(root);
+      argsToDecompile.add("--silent");// why is this required?
+      argsToDecompile.addAll(jarPaths);// all libraries
+      argsToDecompile.add(root);// main source code with issue
       ConsoleDecompiler.main(argsToDecompile.toArray(new String[0]));
       // delete unneccessary legal files
       FileUtils.deleteDirectory(new File(root + "META-INF"));
     }
 
-    // the set of Java classes in the original codebase mapped with their corresponding Java files.
+    /*
+     * ===================================================================================STEP 5: Mapping Classes:====================================================================================
+       >>> WHAT IS A PRMARY CLASS?
+       		A primary class is a class that has the same name as the Java file where the class is declared.	
+       >>> HOW MAPPING WORKS BELOW:		
+       1. Maps Initialization: We initialize two maps: 
+            a. existingClassesToFilePath to keep track of class names and their file paths.
+            b. nonPrimaryClassesToPrimaryClass to map non-primary classes (classes not named after their file) to their primary classes.
+       2. SourceRoot Setup: We use SourceRoot to manage and parse all Java files in the given root directory.
+       3. Parsing Java Files: We call tryToParse() to parse all Java files in the root directory.
+       4. Processing Each Java File:
+          4.1 We get the file path of the current Java file.
+          4.2 We check if the file has a primary type (main class) and get its fully qualified name if it exists.
+          4.3 We loop through all class or interface declarations in the file:
+               4.3.1 If the class has a fully qualified name, we store its name and file path in the existingClassesToFilePath map.
+               4.3.2 If the class is not the primary class, we map it to its primary class in the nonPrimaryClassesToPrimaryClass map.
+          4.4 We also loop through all enum declarations in the file and store their names and file paths in the existingClassesToFilePath map.
+     */
+
     Map<String, Path> existingClassesToFilePath = new HashMap<>();
-    // This map connects the fully-qualified names of non-primary classes with the fully-qualified
-    // names of their corresponding primary classes. A primary
-    // class is a class that has the same name as the Java file where the class is declared.
     Map<String, String> nonPrimaryClassesToPrimaryClass = new HashMap<>();
+    
+ // Initialize SourceRoot to help with parsing the Java files in the specified root directory
```

Comment:
Comments are not necessary when they don't add any information beyond what the code already says. This is an example of such a comment, but most of the one-line comments here (and throughout this PR) have this property. Please go through and remove all of the comments that are duplicative with the code text itself.